### PR TITLE
Run 'go mod verify' at the end of tests

### PR DESCRIPTION
**Pull request by bboreham**
This ensures that dependencies match the list in go.sum.

In particular, this will run as part of CI so we catch cases where vendored files are changed directly.

### CODE REVIEW

File: test_linux.sh [hunk 1]
```diff
@@ -57,3 +57,10 @@ if [ -n "${vetRes}" ]; then
 	echo -e "govet checking failed:\n${vetRes}"
 	exit 255
 fi
+
+echo "Checking go mod verify..."
+modRes=$(go mod verify)
+if [ -n "${modRes}" ]; then
```
Comment (jellonek) [inline test_linux.sh]: The test there should be like:
```sh
if [ $? -gt 0 ]; then
...
```

**Comment by jellonek** [comment]
Related issue: #355

**Comment by squeed** [comment]
It looks like `go mod verify` doesn't do what we want anyways. Instead, we need to "go mod vendor" and see if any files have changed. Bah.

**Comment by bboreham** [comment]
Closing because this isn't actually what we want.

**Pull request closed without merging**